cashweb-registry/src/store/topics.rs

```diff
@@ -82,25 +87,55 @@ impl<'a> DbTopics<'a> {
         if split_topic.iter().any(|segment| segment.is_empty()) {
             return Err(TopicInvalidSegments())?;
         }
-
         let payload_hash = message.payload_hash().as_slice().to_vec();
 
+        // Update existing record if we have one.
+        // TODO: needs a key-level lock on this.
+        let possibly_existing_message = self.get_message(&payload_hash);
```

Comment:
You should add `get_message_opt` which returns `Result<Option<TopicPayload>>`, which is then used in `get_message`, and then you use `get_message_opt(&payload_hash)?` here, followed by a `match` or `if let`